contextualized/regression/datamodules.py

```diff
@@ -0,0 +1,147 @@
+        """
+
+        super().__init__()
+
+        self.dataset = DATASETS[dataset]
+        self.num_workers = 0
+        self.batch_size = 32
+
+        # NOTE: batch size ~ dummy params => each dataset
+        # NOTE: batch size is either too small or
+
+        self.C = torch.tensor(c)
+        self.X = torch.tensor(x)
+        self.Y = torch.tensor(y)
+
+        if correlation or markov:
+            self.Y = self.X
+
+        if n == 0:
+            n = c.shape[0]  # use full dataset
+
+        # partition data
+        train_idx, test_idx, val_idx = self._create_idx(n)
+
+        self.full_dataset = self.dataset(self.C, self.X, self.Y)
+        self.train_dataset = self.dataset(
+            self.C[train_idx], self.X[train_idx], self.Y[train_idx]
+        )
+        self.test_dataset = self.dataset(
+            self.C[test_idx], self.X[test_idx], self.Y[test_idx]
+        )
+        self.val_dataset = self.dataset(
+            self.C[val_idx], self.X[val_idx], self.Y[val_idx]
+        )
+        self.pred_dataset = self.test_dataset  # default to test dataset
+
+    def setup(self, stage: str, pred_dl_type=None):
+        # Assign full/test/train/val datasets for use in dataloaders
+
+        if stage == "predict":
+
+            pred_dl_to_dataset = {
+                "full": self.full_dataset,
+                "train": self.train_dataset,
+                "test": self.test_dataset,
+                "val": self.val_dataset,
+            }
+
+            assert pred_dl_type in [None] + list(
+                pred_dl_to_dataset.keys()
+            ), "Error, invalid dataset type for predict dataloader not specified. Choose from 'test', 'train', 'val', 'full'."
+
+            if pred_dl_type:
+                self.pred_dataset = pred_dl_to_dataset[pred_dl_type]
+
+    def full_dataloader(self):
+        return DataLoader(
+            self.full_dataset,
+            batch_size=self.batch_size,
+            shuffle=True,
+            num_workers=self.num_workers,
+        )
+
+    def train_dataloader(self):
+        return DataLoader(
+            self.train_dataset,
+            batch_size=self.batch_size,
+            shuffle=True,
+            num_workers=self.num_workers,
+        )
+
+    def test_dataloader(self):
+        return DataLoader(
+            self.test_dataset,
+            batch_size=self.batch_size,
+            shuffle=False,
+            num_workers=self.num_workers,
+        )
+
+    def val_dataloader(self):
+        return DataLoader(
+            self.val_dataset,
+            batch_size=self.batch_size,
+            shuffle=False,
+            num_workers=self.num_workers,
+        )
+
+    def predict_dataloader(self):
+        return DataLoader(
+            self.pred_dataset,
+            batch_size=self.batch_size,
+            shuffle=False,
+            num_workers=self.num_workers,
+        )
+
+    def _create_idx(self, n, pct_test=0.2, pct_val=0.3):
```

Comment:
Missed this earlier, but it looks like the test and val splits only exist here, and can't be specified in the `lightning_module.datamodule()` or in the datamodule `__init__` kwargs. Also I'm not sure the `n` kwarg is helpful.